### PR TITLE
4619: Added implementation of GET for commencement date

### PR DESCRIPTION
**Pull request by CallumBradburyBJSS**
Story: [AB#4619](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/4619)
Task: [AB#7198](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7198)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/CommencementDateController.cs [hunk 1]
```diff
@@ -23,9 +23,21 @@ public CommencementDateController(IOrderRepository orderRepository)
         }
 
         [HttpGet]
-        public ActionResult Get()
+        public async Task<ActionResult> Get(string orderId)
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/CommencementDateController.cs:26]: This needs to be changed to GetAsync

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Get Commencement Date.feature [hunk 2]
```diff
@@ -0,0 +1,48 @@
+﻿Feature: Display the Order Commencement Date in a Buyer Section
+    As a Buyer
+    I want to view an order's commencement date
+    So that I can ensure that the information is correct
+    
+Background:
+    Given Orders exist
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Get Commencement Date.feature:7]: I think Sam changed it so you dont need LastUpdated in the table, up to you if you want to remove/keep it

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSteps.cs [hunk 3]
```diff
@@ -222,6 +223,8 @@ private sealed class OrdersTable
 
             public string SupplierContactEmail { get; set; }
 
+            public DateTime? CommencementDate { get; set; }
+
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSteps.cs:227]: New line not needed at the end

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs [hunk 4]
```diff
@@ -63,6 +63,53 @@ public async Task Update_NoOrderFound_ReturnsNotFound()
             result.Should().BeOfType<NotFoundResult>();
         }
 
+        [Test]
+        public async Task Get_WithCommencementDate_ReturnsOkResult()
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs:67]: GetAsync

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs [hunk 5]
```diff
@@ -63,6 +63,53 @@ public async Task Update_NoOrderFound_ReturnsNotFound()
...
+            var model = okResult.Value as CommencementDateModel;
+            model.CommencementDate.Should().Be(context.Order.CommencementDate);
+        }
+
+        [Test]
+        public async Task Get_NullCommencementDate_ReturnsOkResult()
+        {
+            var context = CommencementDateControllerTestContext.Setup();
+            context.Order.CommencementDate = null;
+            var result = await context.Controller.Get("myOrder");
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs:85]: GetAsync

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs [hunk 6]
```diff
@@ -63,6 +63,53 @@ public async Task Update_NoOrderFound_ReturnsNotFound()
...
+            result.Should().BeOfType<OkObjectResult>();
+
+            var okResult = result as OkObjectResult;
+            okResult.Value.Should().BeOfType<CommencementDateModel>();
+            var model = okResult.Value as CommencementDateModel;
+            model.CommencementDate.Should().Be(context.Order.CommencementDate);
+        }
+
+        [Test]
+        public async Task Get_NullCommencementDate_ReturnsOkResult()
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs:81]: GetAsync

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs [hunk 7]
```diff
@@ -63,6 +63,53 @@ public async Task Update_NoOrderFound_ReturnsNotFound()
...
+            result.Should().BeOfType<OkObjectResult>();
+
+            var okResult = result as OkObjectResult;
+            okResult.Value.Should().BeOfType<CommencementDateModel>();
+            var model = okResult.Value as CommencementDateModel;
+            model.CommencementDate.Should().Be(context.Order.CommencementDate);
+        }
+
+        [Test]
+        public async Task Get_OrderNotFound_ReturnsNotFound()
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs:95]: GetAsync

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs [hunk 8]
```diff
@@ -63,6 +63,53 @@ public async Task Update_NoOrderFound_ReturnsNotFound()
...
+        {
+            var context = CommencementDateControllerTestContext.Setup();
+            context.Order = null;
+            var result = await context.Controller.Get("myOrder");
+            result.Should().BeOfType<NotFoundResult>();
+        }
+
+        [Test]
+        public async Task Get_InvalidPrimaryOrganisationId_ReturnsForbid()
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs:104]: GetAsync

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs [hunk 9]
```diff
@@ -63,6 +63,53 @@ public async Task Update_NoOrderFound_ReturnsNotFound()
             result.Should().BeOfType<NotFoundResult>();
         }
 
+        [Test]
+        public async Task Get_WithCommencementDate_ReturnsOkResult()
+        {
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs:68]: Could you make the two Gets with OkResult move into one test and use TestCases, as they are very similar

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs [hunk 10]
```diff
@@ -63,6 +63,53 @@ public async Task Update_NoOrderFound_ReturnsNotFound()
...
+            var result = await context.Controller.Get("myOrder");
+            result.Should().BeOfType<NotFoundResult>();
+        }
+
+        [Test]
+        public async Task Get_InvalidPrimaryOrganisationId_ReturnsForbid()
+        {
+            var context = CommencementDateControllerTestContext.Setup();
+            context.Order.OrganisationId = Guid.NewGuid();
+            context.Order.CommencementDate = DateTime.Now;
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/CommencementDateControllerTests.cs:108]: You also dont have a test that checks if a null parameter is passed into the constructor it throws. Can you add one
